### PR TITLE
feat: add `RankingQuestion` in the Python client

### PR DESCRIPTION
**Pull request by alvarobartt**
# Description

This PR includes the `RankingQuestion`, introduced in #3232, in the Python client, to allow users to create datasets with `RankingQuestion`s and to submit responses for those too.

## Usage

```python
import argilla as rg

rg.init(
    api_url="<ARGILLA_API_URL>",
    api_key="<ARGILLA_API_KEY>,
)

ds = rg.FeedbackDataset(
    fields=[
        rg.TextField(name="prompt-1"),
        rg.TextField(name="prompt-2"),  
    ],
    questions=[
        rg.RankingQuestion(
            name="prompt-ranking",
            description="Rank the prompts from most to least natural.",
            required=True,
            values=["prompt-1", "prompt-2"],
        ),
    ],
)

ds.add_records(
    [
        rg.FeedbackRecord(
            fields={
                "prompt-1": "Explain to a broad audience why banana bread is so fluffy.",
                "prompt-2": "Explain banana banana banana.",
            },
            responses=[
                {
                    "values": {
                        "prompt-ranking": {"value": [{"value": "prompt-1", "rank": 1}, {"value": "prompt-2", "rank": 2}]},
                    },
                    "status": "submitted"
                },
            ],
        ), 
    ]
)

ds.push_to_argilla(name="new-dataset", workspace="new-workspace")
ds = rg.FeedbackDataset.from_argilla(name="new-dataset", workspace="new-workspace")

ds.push_to_huggingface(repo_id="<HUGGINGFACE_REPO_ID>", token="<HUGGINGFACE_TOKEN>")
ds = rg.FeedbackDataset.from_huggingface(repo_id="<HUGGINGFACE_REPO_ID>", token="<HUGGINGFACE_TOKEN>")
```

**Type of change**

- [X] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [X] Added unit tests to cover the new `RankingQuestion`

**Checklist**

- [X] I added relevant documentation
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [X] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/feedback/dataset.py [hunk 1]
```diff
@@ -777,13 +782,15 @@ def format_as(self, format: Literal["datasets"]) -> "Dataset":
                     value = Value(dtype="string")
                 elif question.settings["type"] == "rating":
                     value = Value(dtype="int32")
-                elif question.settings["type"] == "multi_label_selection":
+                elif question.settings["type"] == "ranking":
+                    value = Sequence({"rank": Value(dtype="int32"), "value": Value(dtype="string")})
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset.py]: maybe the `rank` can be an `int16` or even `int8`

File: src/argilla/client/feedback/schemas.py [hunk 2]
```diff
@@ -423,8 +432,38 @@ class MultiLabelQuestion(_LabelQuestion):
     settings: Dict[str, Any] = Field({"type": "multi_label_selection"})
 
 
+class RankingQuestion(QuestionSchema):
+    """Schema for the `RankingQuestion` question-type.
+
+    Args:
+        settings: The settings for the question, including the type and options.
+        values: The values for the question, to be formatted and included as part of
+            the settings.
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/schemas.py:437]: can we add here an example like in the rest of questions?

**Review by gabrielmbmb — APPROVED** [review]
LGTM! Just two minor comments

File: src/argilla/client/feedback/dataset.py [hunk 3]
```diff
@@ -234,10 +235,12 @@ def __init__(
         any_required = False
         unique_names = set()
         for question in questions:
-            if not isinstance(question, (TextQuestion, RatingQuestion, LabelQuestion, MultiLabelQuestion)):
+            if not isinstance(
+                question, (TextQuestion, RatingQuestion, LabelQuestion, MultiLabelQuestion, RankingQuestion)
```
Comment (frascuchon) [inline src/argilla/client/feedback/dataset.py]: Maybe we can externalize this tuple as `SUPPORTED_QUESTION_TYPES` constant and reuse in different places

File: src/argilla/client/feedback/dataset.py [hunk 4]
```diff
@@ -234,10 +235,12 @@ def __init__(
...
+            ):
                 raise TypeError(
                     "Expected `questions` to be a list of `TextQuestion`, `RatingQuestion`,"
-                    " `LabelQuestion`, and/or `MultiLabelQuestion` got a"
+                    " `LabelQuestion`, `MultiLabelQuestion`, and/or `RankingQuestion` got a"
```
Comment (frascuchon) [inline src/argilla/client/feedback/dataset.py]: Here, for example:

```python
f"Expected `questions` to be a list of {SUPPORTED_QUESTION_TYPES!r}..."
```

**Review by frascuchon — APPROVED** [review]

File: src/argilla/client/feedback/dataset.py (hunk 1 above)
Comment (alvarobartt) [reply src/argilla/client/feedback/dataset.py]: Agree, but we should use an unsigned type instead, as ranks are from 1 to N, we can set it to `uint8` 👍🏻

Comment (alvarobartt) [reply src/argilla/client/feedback/dataset.py]:
```suggestion
                    value = Sequence({"rank": Value(dtype="uint8"), "value": Value(dtype="string")})
```

File: src/argilla/client/feedback/dataset.py (hunk 3 above)
Comment (alvarobartt) [reply src/argilla/client/feedback/dataset.py]: We already have `AllowedQuestionTypes`, but that's a `typing.Union` so we cannot re-use it, but yes, we can create a list of types and re-use it here

File: src/argilla/client/feedback/schemas.py (hunk 2 above)
Comment (alvarobartt) [reply src/argilla/client/feedback/schemas.py:437]: Sure thing!

**Comment by alvarobartt** [comment]
Failing tests are unrelated (as most of the times), I guess we could try to work on a PR to add HTTP retries, or split the integration tests from the unit tests to avoid the CI/CD runs from failing so much, WDYT @frascuchon @gabrielmbmb?

File: src/argilla/client/feedback/config.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        ...             visible_labels=None
+        ...         ),
+        ...         rg.MultiLabelQuestion(
+        ...             name="content_class",
+        ...             title="Does the response include any of the following?",
+        ...             description="Select all that apply",
+        ...             labels={"cat-1": "Category 1" , "cat-2": "Category 2"},
+        ...             required=False,
+        ...             visible_labels=4
+        ...         ),
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/config.py:62]: do we want to add here a `rg.RankingQuestion`? is the only one missing

**Review by gabrielmbmb — APPROVED** [review]

File: src/argilla/client/feedback/config.py (hunk 5 above)
Comment (alvarobartt) [reply src/argilla/client/feedback/config.py:62]: Indeed I think we should shorten the rest of them, as we have a lot of information that ends up being more than the actual code and it's hard to navigate through it, I think we can tackle the docstrings in the next release and use a clearer approach

**Pull request merged**